homeassistant/components/unifi/__init__.py

```diff
@@ -64,6 +66,21 @@ async def async_unload_entry(hass, config_entry):
     return await controller.async_reset()
 
 
+async def async_migrate_entry(hass, config_entry):
+    """Migrate old entry."""
+    LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    #  Flatten configuration but keep old data if user rollbacks HASS prior to 2021.03
+    if config_entry.version == 1:
+        data: dict = {**config_entry.data, **config_entry.data[CONF_CONTROLLER]}
+        if config_entry.data != data:
+            hass.config_entries.async_update_entry(config_entry, data=data)
+
+    LOGGER.info("Migration to version %s successful", config_entry.version)
```

Comment:
Since the version is not bumped after migration, maybe decrease severity to debug here, or this will be printed at every setup of unifi.